airflow/gcp/hooks/dataproc.py

```diff
@@ -23,209 +23,47 @@
 
 import time
 import uuid
-from typing import Any, Dict, Iterable, List, Optional
+import warnings
+from typing import Any, Dict, Iterable, List, Optional, Sequence, Tuple, Union
 
-from googleapiclient.discovery import build
-from zope.deprecation import deprecation
+from cached_property import cached_property
+from google.api_core.retry import Retry
+from google.cloud.dataproc_v1beta2 import (  # pylint:disable=no-name-in-module
+    ClusterControllerClient, JobControllerClient, WorkflowTemplateServiceClient,
+)
+from google.cloud.dataproc_v1beta2.types import (  # pylint:disable=no-name-in-module
+    Cluster, Duration, FieldMask, Job, JobStatus, WorkflowTemplate,
```

Comment:
I think `JobStatus` should lead to `enums`, I am not sure about the rest of the types.
There is a lot of mistakes in the `google.cloud` documentation with regard to `types` and `enums`:
https://github.com/PolideaInternal/airflow/pull/279#discussion_r320432094

https://github.com/googleapis/google-cloud-python/issues/7503
https://github.com/googleapis/google-cloud-python/pull/8779
https://github.com/googleapis/google-cloud-python/pull/9165

Reply:
I prefere `types` as this is how documentation describe it, so it's easier to connect docs and code. Not everyone knows about enums and problems you mentioned.